components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala

```diff
@@ -258,6 +266,65 @@ case object LiteralParameterValidator {
 
 }
 
+case class ValidationExpressionParameterValidator(
+    validationExpression: ApiExpression,
+    validationFailedMessage: Option[String]
+) extends ParameterValidator {
+
+  override def isValid(paramName: String, expression: Expression, value: Any, label: Option[String])(
+      implicit nodeId: NodeId
+  ): Validated[PartSubGraphCompilationError, Unit] = {
+    val context = Context("validator", Map(paramName -> value), None)
+    Try(validationExpression.evaluate[Boolean](context, Map())).fold(
+      e =>
+        invalid(
+          CustomParameterValidationError(
+            s"Evaluation of validation expression '${validationExpression.original}' of language ${validationExpression.language} failed: ${e.getMessage}",
+            s"Please provide value that satisfies the validation expression '${validationExpression.original}'",
+            paramName,
+            nodeId.id
+          )
+        ),
+      result => if (result) valid(()) else invalid(error(paramName, nodeId.id))
+    )
+  }
+
+  private def error(paramName: String, nodeId: String): CustomParameterValidationError = CustomParameterValidationError(
+    validationFailedMessage.getOrElse(
+      s"This field has to satisfy the validation expression '${validationExpression.original}'"
+    ),
+    s"Please provide value that satisfies the validation expression '${validationExpression.original}'",
+    paramName,
+    nodeId
+  )
+
+}
+
+object ValidationExpressionParameterValidator {
+
+  implicit val encoder: Encoder[ValidationExpressionParameterValidator] = deriveEncoder
+  implicit val decoder: Decoder[ValidationExpressionParameterValidator] = deriveDecoder
+
+  implicit val apiExpressionEncoder: Encoder[ApiExpression] = {
+    Encoder.forProduct2("language", "original")(e => (e.language, e.original))
+  }
+
+  implicit val apiExpressionDecoder: Decoder[ApiExpression] = {
+    Decoder.forProduct2[ApiExpression, String, String]("language", "original")((l, o) => {
+      new ApiExpression() {
```

Comment:
Why not to throw expression immediately instead of creating failing ApiExpression?

Reply:
This way Nu app will still start, despite having bad config? Or exception will be catched anyway so there is no point of doing this?

Reply:
Why it will start? Why expception will be catched? Who catch it? What is a point of creating fake object? Throwing an exception instead of this is less line of code.